# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [30]:
from IPython.display import Markdown, display

display(Markdown("""
### My Lane

My lane is identifying web pages that are likely to decline in search performance and should be refreshed first.

I chose this lane because it helps prioritize content updates using data. Instead of updating every page, machine learning can help identify pages that are more likely to need attention. This makes the content refresh process more efficient and supports better decision-making.
"""))


### My Lane

My lane is identifying web pages that are likely to decline in search performance and should be refreshed first.

I chose this lane because it helps prioritize content updates using data. Instead of updating every page, machine learning can help identify pages that are more likely to need attention. This makes the content refresh process more efficient and supports better decision-making.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [31]:
from IPython.display import Markdown, display

display(Markdown("""
## The Question: Decision, Action, Cost of a Wrong Call

**Decision:**
The decision is to identify which web pages should be refreshed first based on their likelihood of declining search performance.

**Action:**
The SEO or content team can use these recommendations to prioritize content updates and improve website performance.

**Cost of a Wrong Recommendation:**
If the model recommends the wrong pages, time and resources may be wasted on pages that do not need updates, while important declining pages may be ignored. This can reduce the effectiveness of the content refresh strategy.
"""))


## The Question: Decision, Action, Cost of a Wrong Call

**Decision:**  
The decision is to identify which web pages should be refreshed first based on their likelihood of declining search performance.

**Action:**  
The SEO or content team can use these recommendations to prioritize content updates and improve website performance.

**Cost of a Wrong Recommendation:**  
If the model recommends the wrong pages, time and resources may be wasted on pages that do not need updates, while important declining pages may be ignored. This can reduce the effectiveness of the content refresh strategy.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [32]:
import pandas as pd

df = pd.read_csv("/content/ml-internship-starter/data/raw/content_refresh_anonymized.csv")

print("Dataset Shape:", df.shape)
print("\nColumns:", len(df.columns))
print("\nTrend Direction Counts:")
print(df["trend_direction"].value_counts())

Dataset Shape: (30000, 44)

Columns: 44

Trend Direction Counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


### Quick Look at the Data

The starter dataset contains **30,000 rows** and **44 columns**. The data includes pages with different trend directions. Among them, **16,262 pages are marked as "down"**, **5,962 as "stable"**, and **4,388 as "up"** (with additional pages labeled "new" and "flat"). These numbers show that the dataset is large enough and contains meaningful variation, making it suitable for building a machine learning model that supports content refresh decisions.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### Careful words: what I can and can't claim

This project identifies **observed patterns** in the dataset and provides **directional insights** to support content refresh decisions. The results are intended for **decision support** and can help prioritize which web pages should be refreshed first.

However, this project **does not provide causal proof** that refreshing a page will improve search performance. It **cannot predict Google's ranking algorithm** or guarantee future rankings. The recommendations should be used to support human decision-making rather than replace it.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.